request.js

```diff
@@ -92,7 +92,19 @@ function request(requestOptions, callback) {
   debug(requestId, "calledBack:", calledBack)
   var req = transport.request(options, function onResponse(res) {
     metadata = res
-    res.pipe(concat(collect))
+
+    var contentLength
+    try {
+      contentLength = parseInt(res.headers['content-length'], 10)
```

Comment:
Considering `NaN` to be the same as `Content-Length: 0` messes up other tests. Is it a likely scenario that a user agent would send no `Content-Length` header at all?